tmdb api 요청 및 컴포넌트 수정

sjndowair opened the pull request:
### 풀 리퀘스트 타입
> 하나 이상의 PR 타입을 선택해주세요.
- [x] Add Features
- [ ] Delete features
- [ ] Bug fix
- [ ] Chore jobs
- [ ] Update packages

### 반영 브랜치
> 작업한 브랜치와 작업 내역을 반영할 브랜치를 명세해주세요.
> _ex) feat/refactor-layout -> master_
>

### 변경 사항
> 풀 리퀘스트를 요청하는 코드 내용에 대해 간략하게 명세해주세요.
api로 받아온 정보 배치 및 기존 목업 컴포넌트 삭제 
>

### 특이 사항
> 
> 

### 스크린샷
>
>
<img width="1269" alt="image" src="https://github.com/user-attachments/assets/b32bdf41-6b37-4013-9074-280b9c49a9ab">

sjndowair committed:
feat: 공지사항 컴포넌트 생성

공지사항 컴포넌트 생성으로 인한 noticeContainer 폴더 생성 및 index.tsx 컴포넌틑 style.ts 스타일드 컴포넌트 추가

sjndowair committed:
feat: 공지사항 더미데이터 추가

sjndowair committed:
feat: 스타일 변경

스타일 변경으로 인한 global스타일 변경 및 mediaquery 적용을 위한 mediaQuery 상수 변경

sjndowair committed:
feat: svg 아이콘 추가 및  카드 컴포넌트 코드 수정

sjndowair committed:
fix: header 컴포넌트 반응형 기능 구현

태블릿 width 일 경우 GNB 바가 toggle menu로 변경되도록 수정

sjndowair committed:
feat: api 기능 추가

외부 api를 불러오기위한 key 추가

sjndowair committed:
fix: 카드 컴포넌트 디자인 수정

카드 컴포넌트 arrow before, after에서 svg 이미지 변경 및 위치 수정 및 크기 수정

sjndowair committed:
fix: main 스타일 변경

sjndowair committed:
fix: 공지사항 데이터 컴포넌트 명 변경

sjndowair committed:
feat: movie 및 video 타입 추가

sjndowair committed:
feat: api 요청 및 스타일 추가

sjndowair committed:
feat: 이미지 및 동영상 api 키 추가

sjndowair committed:
feat: scroll 기능 추가로 인한 컴포넌트추가

scroll이 최상단에서 벗어날시 오른쪽 하단에 버튼 구현(버튼 클릭 시 scroll 최상단으로 이동하는 이벤트 발생 예정)

sjndowair committed:
fix : svg 파일 위치 변경 및 스타일 추가

sjndowair committed:
feat: footer 메뉴 추가 및 tmdb api 속성 추가

sjndowair committed:
feat: 푸터 컴포넌트 추가

sjndowair committed:
feat: api 요청 주소 추가

sjndowair committed:
fix: 백그라운드 색상 변경 및 mediasize수정

sjndowair committed:
fix: 카드컴포넌트 수정

기존 더미 데이터 코드 삭제 및 오픈 api로 불러온 데이터로 변경

sjndowair committed:
fix: notice 컴포넌트 스타일 수정

svg파일 삭제 및 스타일 수정으로 인한 컴포넌트 수정

sjndowair committed:
delete: 기존 더미데이터 삭제

sjndowair committed:
fix: 푸터 컴포넌트 내용 수정

sjndowair committed:
fix: 메인 컴포넌트 스타일 수정

위치이동

sjndowair committed:
fix: 카드 컴포넌트 타이틀  내용 수정

sjndowair committed:
fix: 메인 케러셀 수정

기존 더미데이터에서 오픈 api 데이터로 교체 및 스타일 수정

sjndowair committed:
fix: header 컴포넌트 디자인 수정

sjndowair committed:
fix: 컴포넌트 분할

header, mainCarousal, cardCarousal, notice, footer 컴포넌트 순으로 분할

sjndowair committed:
feat: 리액트 라우터 설치

sjndowair committed:
fix: TMDB API 코드 마이그레이션

songforthemute reviewed (changes requested):
1. 컴포넌트 인터페이스 정의 및 분리
2. 변수, props, 컴포넌트 이름 다시 한번 더 고려해보기 + 고민 많이하기
3. 불필요한 깊은 depth의 마크업 지양
4. console.log는 테스트할때 열심히 쓰고, 지우기
5. 옵셔널 체이닝 + null 병합 연산자 잘 활용해서 예외처리, 에러처리 잘 하기
6. 배열 내부 요소들도 옵셔널 체이닝 할 수 있다.
7. 상수는 컴포넌트 바깥에 정의하고 가급적이면 분리할 것, 상태로 관리해야하는 요소라면 state로, 동적인 요소를 얻어오는 것에 불과하다면 함수로 분리하는 것을 고려하기
8. 매직 넘버 가급적이면 지양하기
9. 예외처리 잘하기! 꼭!
10. 브라우저 렌더링 파이프라인에 대해서 조금 더 공부하고, 생각해보기

sjndowair committed:
fix: svg 컴포넌트 분리

svg 컴포넌트 스타일 구조분해할당 상수 영역으로 변경

sjndowair committed:
fix: header 영역 수정

깊은 뎁스 삭제 및 depth 스타일 변경 svg파일 위치 변경

sjndowair committed:
fix: notice 컴포넌트 수정

notice 컴포넌트 svg 파일 위치 이동 및 스타일 수정

sjndowair committed:
fix: 카드 컴포넌트 svg파일 이동

sjndowair committed:
fix: 메인케러셀 svg파일 이전

sjndowair committed:
fix: scrollbutton componet 위치변경

sjndowair committed:
add: modal 임시 구현

 useEffect height 이벤트로 변경 예정

sjndowair committed:
feat: 모달 경로 및 메인페이지 게시

sjndowair committed:
fix: 모달 오늘하루보지않기 버튼 활성화

sjndowair committed:
add: subpage 컴포넌트 분리

sjndowair committed:
fix: header 수정

header에서 사용하지 않는 상태 및 프롭삭제

sjndowair committed:
fix: 전역 스타일 변경

sjndowair committed:
add: svg 스타일 추가

기존에 사용하던 heroIcon 1 에서 heroIcon2 변경으로 인한 스타일 적용 및 변경

sjndowair committed:
feat: Card store 추가 ( theme ctx 참조)

sjndowair committed:
feat: tv serise context 추가

sjndowair committed:
fix: subpage 컴포넌트 추가

sjndowair committed:
feat: TMDB 오픈 api추가

sjndowair committed:
fix: 컴포넌트 전체 스타일 수정

sjndowair committed:
fix: 카드 캐러셀 컴포넌트 수정

tv api 및 스타일 수정으로 인한 수정

sjndowair committed:
fix: tvCtx 추가로 인한 컴포넌트 명 수정

기존 cardCarousalCtx 컴포넌트명 movieCtx로 수정 완료

sjndowair committed:
fix: 리액트 라우터 추가로 인한 링크 추가

sjndowair committed:
feat: api추가로 인한 카드캐러셀 추가

sjndowair committed:
fix: notice컴포넌트 스타일 수정

margin-top 추가

sjndowair committed:
feat: tv serise  타입 정의 추가

sjndowair committed:
feat: cardProvider 추가

cardProvider 추가로 인한 수정

songforthemute commented:
```ts
import { useEffect, useState } from "react";

import Layout from "../components/layout"; // 레이아웃 컴포넌트

import styled from "styled-components";

const ModalBackground = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.5);
    z-index: 9999;
`;

const ModalContainer = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background-color: white;
    padding: 20px;
    border-radius: 10px;
    z-index: 10000;
`;

const LOCAL_STORAGE_NOT_TODAY_KEY = "LOCAL_STORAGE_NOT_TODAY_KEY";

const getTodayDate = () => {
    return new Date().toLocaleDateString("ko-kr", {
        dateStyle: "medium",
    });
};

const Home = () => {
    const [isOpenModal, setIsOpenModal] = useState<boolean>(false);
    const [isCheckedNotToday, setIsCheckedNotToday] = useState<boolean>(false);

    const onCloseModal = () => {
        setIsOpenModal(false);

        if (isCheckedNotToday) {
            localStorage.setItem(LOCAL_STORAGE_NOT_TODAY_KEY, getTodayDate());
        }
    };

    useEffect(() => {
        const notTodayDate = localStorage.getItem(LOCAL_STORAGE_NOT_TODAY_KEY);

        if (!notTodayDate || notTodayDate !== getTodayDate()) {
            setIsOpenModal(true);
        }
    }, []);

    return (
        <Layout>
            {isOpenModal && (
                <ModalBackground
                    onClick={(e) => {
                        e.preventDefault();
                        e.stopPropagation();
                        onCloseModal();
                    }}
                >
                    <ModalContainer>
                        <label
                            onClick={(e) => {
                                e.preventDefault();
                                e.stopPropagation();
                                setIsCheckedNotToday((prev) => !prev);
                            }}
                        >
                            <input
                                type="checkbox"
                                checked={isCheckedNotToday}
                            />
                            <span style={{ color: "#000" }}>
                                오늘 하루 보지 않기
                            </span>
                        </label>
                    </ModalContainer>
                </ModalBackground>
            )}
        </Layout>
    );
};
```

songforthemute commented:
### 240802

> 1. svg 조금 더 공통화 _(굳이 styled한 컴포넌트일 필요가 없음)_
> 2. 변수/상수명 오탈자가 여전히 많이 존재함
> 3. 캐러셀 컴포넌트, 다른 컴포넌트도 그렇지만, 데이터는 페이지나 필요한 데에서 부르고, 컴포넌트는 받은 데이터를 잘 그려주고 상호작용에 유연할 수 있도록 분리하는 연습. 페이지와 컴포넌트의 역할 분담/분리.
> 4. 모달 _(cf. https://github.com/sjndowair/movvy/pull/7#issuecomment-2265771119)_
> 5. 디자인에 너무 신경쓰지 말 것. 디자인에 과하게 신경써서 애플리케이션이 빌딩이 안되고 있음.
> 6. 불편하거나 어색한 점이 생기면, 글로 정의하는 연습을 할 것. 그래야 구체적으로 내가 어떤 상황에 처해 있고, 어떤 방법을 쓸 수 있고를 판단하고, 설명할 수 있음.
> 7. 질문 좀 할 것..
> 8. 통일되지 않는 스타일 사용, 과한 스타일 혼용 자제할 것 + 고정 수치 지양하고, 반응형으로 짤 것
>     - ex1. flex와 float 같이 쓰기
>     - ex2. hex code와 rgb code 같이 쓰기
>     - ex3. 56.3vw 같은 이상한 수치 쓰지 않기

sjndowair committed:
fix: svg 컴포넌트 위치 이동

sjndowair committed:
fix: 모달컴포넌트 상태 이동

모달컴포넌트의 상태를 root 컴포넌트에서 가져와 사용

sjndowair committed:
feat: svg컴포넌트 변경으로 인한 경로 변경

sjndowair committed:
fix: svg 컴포넌트 수정

styled 컴포넌트 삭제

sjndowair committed:
fix: 카드케러셀 타이틀평 스타일수정

sjndowair committed:
delete: 필요없는 컴포넌트 및 ctx 삭제

sjndowair committed:
add: tanstackQuery 및 reactQuery추가

sjndowair committed:
feat: reactRouter, HashRouter 로 교체

sjndowair committed:
fix: 카드캐러셀 수정

기존 useEffect로 관리하던 api 상태변경

sjndowair committed:
fix: 일부 컴포넌트  경로 및  타입 지정 수정

sjndowair committed:
feat: useQuery로 api 받아오기 변경

sjndowair committed:
delete: reactQuery 삭제

sjndowair committed:
fix: heingt 이벤트 시 스로틀링 적용

sjndowair committed:
feat: series 컴포넌트추가

sjndowair committed:
fix: 카드 캐러셀 스타일 수정

hover 이벤트 변경

sjndowair committed:
feat: 스로틀링 devDependencies추가

sjndowair committed:
feat: tv series api 추가

sjndowair committed:
fix: 카드 캐러셀 컴포넌트 일부 기능 수정

sjndowair committed:
fix: header router 수정

sjndowair committed:
feat: react-player 추가

sjndowair committed:
fix : 모달 외부 클릭시에도 모달 종료기능 구현

sjndowair committed:
fix: 콜백함수명 수정

sjndowair committed:
feat: video 컴포넌트 추가

메인케러셀 클릭 시 비디오 컴포넌트 노출

sjndowair committed:
fix: maincarousal 데이터 외부로 이동

sjndowair committed:
fix: hashRouter 컴포넌트 이동

sjndowair committed:
fix: imagePath 옵셔널체이닝 추가

sjndowair committed:
feat: 장르 유틸 추가

비디오 컴포넌트 클릭 시 장르 노출하기 위해 유틸 추가

sjndowair committed:
fix:  series 및 movie url 구분을 위한 api수정

sjndowair committed:
feat: video 컴포넌트 추가

sjndowair committed:
feat: api 및 api type 추가

sjndowair committed:
fix: header 컴포넌트 input 추가

sjndowair committed:
feat: detailPage  api 추가

detailPage api 및 api interface 추가

sjndowair committed:
feat: detail page 추가

검색창에 검색어 입력 후 클릭 시 detail page 가 나올수 있도록 page 추가

sjndowair committed:
feat: searchPage 추가

header 검색창 입력 시 searchPage가 나올수 있도록 추가

sjndowair committed:
fix: cardCarousal props, temporary props로 수정

console창 오류로 인한 temporary props로 수정

sjndowair committed:
fix: 검색창 useForm으로 수정

useForm으로 관리 및 onClickMagnifiy 함수에 setTimeout을 사용하여 focus가 되도록 수정 => 기존은 검색창이 열리면 focus가되지않았음

sjndowair committed:
fix: header props, temporary props 로 수정

conosole 오류 삭제를 위한 temporary props 로 수정

sjndowair committed:
fix: series page 수정

useMatch 추가 및 useQuery에서 isLoading을 사용하지 않아 삭제 완료

sjndowair committed:
fix: mainCarousal 클릭 시 videoModal 나오게 수정

 mainCarousal 클릭 시 videoModal 나오게 useNavigate 추가  및 movie data 와 series 데이터 구분을 위한 타입 추가

sjndowair committed:
fix:  series movie id 수정

api가 seires 가 아닌 tv로 경로가 설정되어있어 수정

sjndowair committed:
delete: noticepage 삭제

sjndowair committed:
fix: data undefined, null값으로 수정

sjndowair committed:
delete: video page page에서 component로  수정으로 인한 삭제

sjndowair committed:
fix: router 경로 수정

sjndowair committed:
delete: 더미데이터삭제

sjndowair committed:
fix: seires page useMatch 추가

sjndowair committed:
fix: 카드케러셀 인터렉션 수정

sjndowair committed:
fix: 스터디 수정

sjndowair committed:
fix: header  컴포넌트 수정

스타잃 수정

sjndowair committed:
delete: notice 삭제

sjndowair committed:
feat: root page 위치수정

sjndowair committed:
fix: global style 수정

헥사코드 > rgba

sjndowair committed:
fix: cardCarousal 반응형 수정

반응형시 등장하는 카드 개수 변경

sjndowair committed:
fix:  footer width 추가

sjndowair committed:
fix: Escape key code추가

esc클릭 시 useNavigate가 동작할수있도록 수정

sjndowair committed:
fix: 별점 및 장르 추가

sjndowair committed:
fix: 새로고침 시 promise  Error 반환으로 인한 수정

새로고침시에도 해당 api를 불러올수있도록 수정

sjndowair committed:
fix: 비디오컴포넌트 스타일 반응형 수정

sjndowair committed:
fix: mainCarousal 수정

스타일 수정 및 사용하지않는 props 삭제

sjndowair committed:
fix: useForm 포커스 추가

ref로 와 setTimeout으로 포커스가 향해지던 코드를 useForm으로 수정

sjndowair committed:
fix: header component 스타일 수정

sjndowair committed:
delete: 사용하지않는 layout styledcomponent 삭제

sjndowair committed:
fix: 카드컴포넌트의 상태 추가로 인터렉션 추가

sjndowair committed:
feat: searchPage title추가

sjndowair committed:
fix: router 경로 수정

sjndowair committed:
delete: 사용하지않는 userEvent삭제

sjndowair committed:
fix: detailpage 반응형 스타일 수정

반응형 스타일 수정 및 video가 없을경우 다르게 나올수있도록 수정

sjndowair committed:
delete: hashRouter 위치 수정

sjndowair committed:
feat: react-hook-form추가

sjndowair committed:
fix: 비디오 컴포넌트 발생시 이벤트추가

스크롤락 및 인터렉션 추가

sjndowair committed:
delete: svg 옵션수정

songforthemute committed:
fix: 코멘터리 수정

sjndowair committed:
delete: modal 삭제

videoComponent로 모달창을 보여주기 때문에 필요없는것으로 판단

sjndowair committed:
fix: layout page 위치 수정

sjndowair committed:
fix: mainCarousal component 위치 수정

sjndowair committed:
feat: theme toggle 기능 추가

sjndowair committed:
fix: root page 변경

root page 명 변경 및 폴더 이동 로직 수정

sjndowair committed:
delete: 사용하지않는 devtool 삭제

sjndowair committed:
fix: 컴포는 interface 추가

sjndowair committed:
fix: cardCarousal 수정

스타일 비율 수정 및 로직 수정 useCallback 사용 시  window innerwidth 가 변경되지않으면 캐러셀 동작 하지않아 수정

sjndowair committed:
fix: header style 수정

sjndowair committed:
delete: svg styled 컴포넌트 삭제

styled 텀포넌트 사용 시 유지보수에 용이하지않아 svg interface로 속성을 주어 직접 하나의 컴포넌트 안에서 수정할수있는게 용이 하다 판단 하여 수정

sjndowair committed:
fix: video component  수정

react portal을 활용하여 동작하도록 로직 수정

sjndowair committed:
fix: series page 위치 수정

sjndowair committed:
fix: scrollButton 위치 수정

sjndowair committed:
feat: custom hook 추가

sjndowair committed:
fix: detail page 수정

detail page 폴더 이동 및 컴포넌트 명 수정, 스타일 수정

sjndowair committed:
fix: search page 수정

seach page 폴더 이동 및 컴포넌트 명 변경, 스타일 수정

sjndowair committed:
fix: 사용 하지 않는 import 삭제

sjndowair committed:
fix: 리액트 쿼리 컴포넌트 분리

sjndowair committed:
fix: root page style themeCtx로 수정

sjndowair committed:
fix: dark 모드 수정

localstorage 롤 페이지 이탈 시에도 마지막에 선택 해둔 테마로 유지되도록 로직 수정

sjndowair committed:
fix: theme 수정

sjndowair committed:
feat: detail 페이지 isDark props 및 스타일 수정

sjndowair committed:
fix: video util 수정

video util url 수정

sjndowair committed:
delete: 사용하지않는 모달 유틸 삭제

sjndowair committed:
fix: detail page 수정

코드 수 축소

sjndowair committed:
fix: 이중 부정 연산자로 수정

sjndowair committed:
fix: json 변경

sjndowair commented:
최종 수정
